lib/Myriad/API.pm

```diff
@@ -45,7 +50,7 @@ This can be used to call RPC methods and act on subscriptions.
 =cut
 
 method service_by_name ($name) {
-    return $myriad->service_by_name($name);
+    return Myriad::Service::Remote->new(service_name => $myriad->registry->make_service_name($name), myriad => $myriad);
```

Comment:
would likely need to abstract this out rather than creating the class directly, but maybe not needed right away